src/data/posts.js

```diff
@@ -235,3 +235,11 @@ export const QUERY_POST_SEO_BY_SLUG = gql`
     }
   }
 `;
+
+export const QUERY_POST_PER_PAGE = gql`
```

Comment:
just out of curiosity, do you think this is better suited as its own individual query?

wouldn't we have it available throughout the application to easily use if we added it to the `site` query?

https://github.com/colbyfayock/next-wordpress-starter/blob/76e4f017c92107945171713f7b9c3a4910c11cd6/src/data/site.js#L3

thoughts?

Reply:
ah thats a really good idea! totally missed.  I will make the changes, thanks @colbyfayock !

Reply:
Having a problem with this approach: If we store this value withing the site context, we can't have access to it inside fetching functions (getStaticProps) to pre-render as is required to generate the pagination ... Do you know if there is a way to retrieve this context data inside getStaticProps? Maybe in this case its worth having like its now?

Reply:
so interestingly, i was using NextAuth which passes looks at the `context` available as an arugment on `getServerSideProps`, i wonder if it's available on there by change?

https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation

i'm not sure how NextAuth is doing it though so might be a long shot

Reply:
I think they work with cookies: when using **getServerSideProps** is possible to extract from that context object the http request and check the cookie with the info. In the case of **getStaticProps** is [not possible](https://nextjs.org/docs/basic-features/data-fetching#only-runs-at-build-time).... trying to figure out if there is another way to do it ..